lib/sycamore/sycamore/connectors/neo4j/neo4j_writer.py

```diff
@@ -58,10 +60,12 @@ def create_target_idempotent(self, target_params: BaseDBWriter.TargetParams):
         with self._driver.session(database=target_params.database):
             pass
 
-    def _write_nodes_neo4j(self, nodes: list[str], session: "Session"):
+    def _write_nodes_neo4j(self, nodes: list[tuple[str, str]], session: "Session"):
         for node_type in nodes:
-            file_url = f"file:///sycamore/nodes/{urllib.parse.quote(node_type)}"
-            node_label = node_type[:-4]
+            node_label = node_type[0]
+            file_url = node_type[1]
+            if "s3.amazonaws.com" not in file_url:
+                file_url = f"file:///sycamore/nodes/{urllib.parse.quote(file_url)}"
```

Comment:
ohhhh yes the name is quite bad i'll change.

For the `s3.amazonaws.com` check its because signed urls are not aws uri's they look like this

`https://mybucket.s3.amazonaws.com/myfile.txt?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=YOUR_AWS_ACCESS_KEY_ID%2F20231101%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20231101T120000Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=EXAMPLE_SIGNATURE`